web/src/app/components/map/map.component.ts

```diff
@@ -411,77 +429,40 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     marker.setIcon(newIcon);
   }
 
-  private selectPolygon(polygon: google.maps.Polygon | undefined) {
-    if (polygon === this.selectedPolygon) {
-      return;
+  private selectPolygons(locationOfInterestId: string | null) {
+    const polygons = locationOfInterestId
+      ? this.polygons.get(locationOfInterestId)
+      : undefined;
+    if (polygons) {
+      for (const polygon of polygons) {
+        polygon.setOptions({strokeWeight: enlargedPolygonStrokeWeight});
+      }
     }
-    if (polygon) {
-      polygon.setOptions({strokeWeight: enlargedPolygonStrokeWeight});
+    const selectedPolygons = this.selectedLocationOfInterestId
+      ? this.polygons.get(this.selectedLocationOfInterestId)
+      : undefined;
+    if (selectedPolygons) {
+      for (const polygon of selectedPolygons)
+        polygon.setOptions({
+          strokeWeight: normalPolygonStrokeWeight,
+        });
     }
-    if (this.selectedPolygon) {
-      this.selectedPolygon.setOptions({
-        strokeWeight: normalPolygonStrokeWeight,
-      });
-    }
-    this.selectedPolygon = polygon;
-    this.panAndZoom(polygon?.getPaths().getAt(0).getAt(0));
-  }
-
-  private addGeoJsonLocationOfInterest(
-    color: string | undefined,
-    jobName: string | undefined,
-    loi: GeoJsonLocationOfInterest
-  ) {
-    const paths: google.maps.LatLng[][] = [];
-    const job = new google.maps.Data();
-    job.addGeoJson(loi.geoJson);
-    job.forEach(f => {
-      if (f.getGeometry() instanceof google.maps.Data.Polygon) {
-        paths.push(
-          ...this.geoJsonPolygonToPaths(
-            f.getGeometry() as google.maps.Data.Polygon
-          )
-        );
-      }
-      if (f.getGeometry() instanceof google.maps.Data.MultiPolygon) {
-        (f.getGeometry() as google.maps.Data.MultiPolygon)
-          .getArray()
-          .forEach(polygon =>
-            paths.push(...this.geoJsonPolygonToPaths(polygon))
-          );
-      }
-    });
-
-    this.addPolygonToMap(loi.id, color, jobName, paths);
-  }
-
-  private geoJsonPolygonToPaths(
-    polygon: google.maps.Data.Polygon
-  ): google.maps.LatLng[][] {
-    const paths: google.maps.LatLng[][] = polygon
-      .getArray()
-      .map(linearRing => linearRing.getArray());
-    return paths;
-  }
-
-  private addAreaOfInterest(
-    color: string | undefined,
-    jobName: string | undefined,
-    loi: AreaOfInterest
-  ) {
-    const vertices: google.maps.LatLng[] = loi.polygonVertices.map(
-      vertex => new google.maps.LatLng(vertex.latitude, vertex.longitude)
-    );
-
-    this.addPolygonToMap(loi.id, color, jobName, [vertices]);
+    const firstPolygon = polygons && polygons[0];
+    this.panAndZoom(firstPolygon?.getPaths().getAt(0).getAt(0));
```

Comment:
Sure, that works for me! I can see the advantage of focusing in on the location, but I don't think it does too much at the moment, especially since we're not focusing in on the entire multipolygon.